src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -74,6 +74,8 @@ internal class MsalAccessTokenCacheItem : MsalCredentialCacheItemBase
             }
 
             HomeAccountId = homeAccountId;
+
+            AddJitterToTokenExpiration();

```

Comment:
@bgavrilMS The expiration is used to calculate the suggestedCacheExpiry which happens when the token is saved to the cache. If the jitter is added when the token is retrieved from the cache the suggestedCacheExpiry will no longer be accurate, although the difference will be within 5 mins. Thus, I added the jitter modification in this constructor.

Thoughts?

Reply:
1. Let's only modify RefreshIn
2. MsalAccessTokenCacheItem object is a data object. It should not know about jitter and other concepts like this. By adding jitter here, it's easy to make a mistake like - add jitter on token response, then later load the cache -> add jitter again.

Probably the simplest approach here is to change the refresh_in value from MSAL token response.